### PR TITLE
Feature/analytics ui logging

### PR DESCRIPTION
**Pull request by alfredovargasjr**
# Description

Added logging to the UI. Only goes to the controller level, no data store.

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

# Notes

Put any notes here

# Pivotal Tracker Link:

PT link here

### CODE REVIEW

File: Backend/DTO/LoggingAPI/LoggingRequests.cs [hunk 1]
```diff
@@ -0,0 +1,37 @@
+﻿using System;
+using System.Collections.Generic;
```
Comment (ChrisMeyer7088) [inline Backend/DTO/LoggingAPI/LoggingRequests.cs:2]: What is the point of these DTOs?

File: Backend/DTO/LoggingAPI/LoggingResponses.cs [hunk 2]
```diff
@@ -0,0 +1,12 @@
+﻿using System;
```
Comment (ChrisMeyer7088) [inline Backend/DTO/LoggingAPI/LoggingResponses.cs]: Purpose of this DTO?

File: Backend/DTO/LoggingAPI/LoggingRequests.cs (hunk 1 above)
Comment (ChrisMeyer7088) [reply Backend/DTO/LoggingAPI/LoggingRequests.cs:2]: Is this the object we will be receiving from the Frontend?

File: Frontend/src/views/Account.vue [hunk 3]
```diff
@@ -63,11 +64,24 @@ export default {
...
+          webpageDurationStart: 0,
+        }
       }
   },
   created() {
+    // Browser logger, listener if used switches/closes tab
+    document.addEventListener('beforeunload', this.userBrowserTabSession)
+    this.logging.webpage = this.$options.name;
+    this.logging.webpageDurationStart = Date.now();
+    window.addEventListener('beforeunload', this.userBrowserTabSession)
```
Comment (ChrisMeyer7088) [inline Frontend/src/views/Account.vue]: Does the window event listener get called if the url changes?

File: Frontend/src/views/Account.vue [hunk 4]
```diff
@@ -63,11 +64,24 @@ export default {
...
     this.getUserAccount();
   },
+  destroyed() {
```
Comment (ChrisMeyer7088) [inline Frontend/src/views/Account.vue:79]: Is destroy called when the browser window is closed, if so no need for eventlisteners?

File: Frontend/src/views/Account.vue [hunk 5]
```diff
@@ -63,11 +64,24 @@ export default {
...
+        logging: {
+          webpage: '',
+          webpageDurationStart: 0,
+        }
       }
   },
   created() {
+    // Browser logger, listener if used switches/closes tab
+    document.addEventListener('beforeunload', this.userBrowserTabSession)
+    this.logging.webpage = this.$options.name;
```
Comment (ChrisMeyer7088) [inline Frontend/src/views/Account.vue:75]: $options is the export default I take it?

File: Frontend/src/views/AdminDashboard.vue [hunk 6]
```diff
@@ -44,30 +44,41 @@ import axios from 'axios'
...
+    DashboardMenuItems: [
+        {title: 'User Management', icon: 'account_box'},
+        {title: 'Analytics', icon: 'timeline'},
+        {title: 'App Publish', icon: 'build'}
+    ],
+    logging: {
+      webpageDurationStart: 0,
     }
+  }),
+  created() {
```
Comment (ChrisMeyer7088) [inline Frontend/src/views/AdminDashboard.vue:69]: Didn't create eventlisteners on the other pages, so I take it destroyed handles when the browser window closes?

File: Frontend/src/views/Account.vue (hunk 3 above)
Comment (ChrisMeyer7088) [reply Frontend/src/views/Account.vue]: I think the 2 event listeners and destoryed function will make 3  separate calls to userBrowserTabSession

File: Backend/DTO/LoggingAPI/LoggingRequests.cs (hunk 1 above)
Comment (alfredovargasjr) [reply Backend/DTO/LoggingAPI/LoggingRequests.cs:2]: yes

File: Backend/DTO/LoggingAPI/LoggingResponses.cs (hunk 2 above)
Comment (alfredovargasjr) [reply Backend/DTO/LoggingAPI/LoggingResponses.cs]: none lol

File: Frontend/src/views/Account.vue (hunk 5 above)
Comment (alfredovargasjr) [reply Frontend/src/views/Account.vue:75]: yes

**Review by julianpoy — APPROVED** [review]

**Pull request merged**